tests/test_gpu_examples.py

```diff
@@ -1930,6 +1931,200 @@ def test_causal_lm_training_aqlm(self):
             assert trainer.state.log_history[-1]["train_loss"] is not None
 
 
+@require_torch_gpu
+@require_hqq
+@unittest.skipUnless(
+    version.parse(importlib.metadata.version("transformers")) >= version.parse("4.36.1"),
+    "test requires `transformers>=4.36.1`",
+)
+class PeftHqqGPUTests(unittest.TestCase):
+    r"""
+    HQQ + peft tests
+    """
+
+    def setUp(self):
+        self.causal_lm_model_id = "TinyLlama/TinyLlama-1.1B-Chat-v1.0"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.causal_lm_model_id)
+
+    def tearDown(self):
+        r"""
+        Efficient mechanism to free GPU memory after each test. Based on
+        https://github.com/huggingface/transformers/issues/21094
+        """
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    @pytest.mark.single_gpu_tests
+    @parameterized.expand([False, True])
+    def test_causal_lm_training_hqq(self, use_dora):
+        r"""
+        Test the CausalLM training on a single GPU device. The test would simply fail if the adapters are not set
+        correctly.
+        """
+
+        use_transformers = False
+        try:
+            from transformers import HQQConfig
+
+            use_transformers = True
+        except ImportError:
+            from hqq.core.quantize import BaseQuantizeConfig
+            from hqq.engine.hf import HQQModelForCausalLM
+
```

Comment:
Can you instead split the test into two ? In any case we do install transformers from source in the docker image !